# App revenue analysis (project1)
This program uses official Data from the AppleStore. All Data is stored in the "AppleStore.csv" file.
The goal is, to analyse which app-type/genre , makes the most profit.
With that infromation, our developers can build an App with good odds, that the app will be a success. 

Data:

**"googleplaystore_project1.csv"** [LINK] (https://www.kaggle.com/lava18/google-play-store-apps)
10,000 Android apps from Google Play; the data was collected in August 2018 

**"AppleStore_project1.csv"** [LINK] (https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)
7,000 iOS apps from the App Store; the data was collected in July 2017 

Source of the csv files: Dataquest.io

In [1]:
opened_file_apple = open('AppleStore_project1.csv', encoding="utf8")
opened_file_google = open('googleplaystore_project1.csv', encoding="utf8")
from csv import reader
read_file_apple = reader(opened_file_apple)
read_file_google = reader(opened_file_google)
apps_data_apple = list(read_file_apple)
apps_data_google = list(read_file_google)
opened_file_apple.close()
opened_file_google.close()

# function from Dataquest 
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        
explore_data(apps_data_apple,0,2)
explore_data(apps_data_google,0,2)




['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']




# Data Cleaning

 ## Missing values
 This function i wrote, checks if the length if an sublist is not equal the length of the header sublist.
 It returns a list, with all index'es where the length is not how it should be. 
 This implies, that there is data missing, or too much datapoints.
 To check that, you'll have to look at that index manually.
 I see that in the AppleStore Dataset is something missing at Index 10473.
 I delete the whole row. 

In [2]:
def list_length_check(dataset):
    header_length=len(dataset[0])
    missing_values_index=[]
    for counter, value in enumerate(dataset):
        if len(value)!=header_length:
            missing_values_index.append(counter)
        
    if missing_values_index==[]:
        print("Check! All good!")
    else: print("Length not correct at index: "+str(missing_values_index))
                                
#---------------------------------------------------------------------

print("Google:") #Fehler bei Index 10473
list_length_check(apps_data_google)
print("\n")
print("Apple:") #Alles gut
list_length_check(apps_data_apple)

del apps_data_google[10473] # Löschen der Fehlerhaften Zeile !ONLY RUN THIS ONCE! 
del apps_data_google[10200]        


Google:
Length not correct at index: [10473]


Apple:
Check! All good!


# Duplicate remover function
Here i use my function only on the google dataset. Since the apple dataset doesnt have any duplicate data. 

In [3]:
def check_for_duplicates_v2(dataset,title_column,rating_column):

    global unique_data 
    rating_dict={}
    unique_data=[dataset[0]] #Header
    already_added=[]
    for data in dataset[1:]:
        title=data[title_column]
        rating=data[rating_column]
        if title in rating_dict and rating > rating_dict[title]: ## nur in dict geschrieben wenn rating aus dataset größer oder name noch nicht im dict##
            rating_dict[title]=rating
        elif title not in rating_dict:
            rating_dict[title]=rating
            
    
    for data in dataset[1:]:
        title=data[title_column]
        rating=data[rating_column]
        if (rating_dict[title]==rating) and (title not in already_added):
            unique_data.append(data)
            already_added.append(title)
    print("DICT ist:", len(rating_dict))
    
        
        

###google###    
check_for_duplicates_v2(apps_data_google,0,3)
google_unique=unique_data

apple_unique=apps_data_apple
print("Length GOOGLE: ", len(google_unique))


print(google_unique[:2])


DICT ist: 9658
Length GOOGLE:  9659
[['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'], ['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']]


# My own duplicate checker
Checks if the (name and the review count) of a sublist exists more than once. 

In [4]:
test_list=[['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic'], ['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'], ['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1'], ['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1'], ['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1'], ['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1'], ['429047995', 'Pinterest', '74778624', 'USD', '0.0', '1061624', '1814', '4.5', '4.0', '6.26', '12+', 'Social Networking', '37', '5', '27', '1'], ['282935706', 'Bible', '92774400', 'USD', '0.0', '985920', '5320', '4.5', '5.0', '7.5.1', '4+', 'Reference', '37', '5', '45', '1'], ['553834731', 'Candy Crush Saga', '222846976', 'USD', '0.0', '961794', '2453', '4.5', '4.5', '1.101.0', '4+', 'Games', '43', '5', '24', '1'], ['324684580', 'Spotify Music', '132510720', 'USD', '0.0', '878563', '8253', '4.5', '4.5', '8.4.3', '12+', 'Music', '37', '5', '18', '1']]
test_list.append(['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'])
test_list.append(['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974677', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'])

def check_for_duplicates_own(dataset,title_column,rating_column):

    unique=[]
    duplicate=[]
    for i, data in enumerate(dataset):
        if [data[title_column],data[rating_column]] in unique:
            duplicate.append([data[title_column],data[rating_column]])
        else:
            unique.append([data[title_column],data[rating_column]]) 
            
    for i, data in enumerate(duplicate):
        for i2, data2 in enumerate(dataset):
            value_pair=[data2[title_column],data2[rating_column]]
            if value_pair == data:
                print(data2[title_column],"at Index:",i2)

###google###    
print("100% duplicates GOOGLE:\n")
check_for_duplicates_own(google_unique,0,3)

print("100% duplicates APPLE:\n")
check_for_duplicates_own(apple_unique,1,5)










100% duplicates GOOGLE:

100% duplicates APPLE:



# Remove all non-roman letters, the way Dataquest teaches us
Since my take above worked, but filtered out too much, so my function wasnt really useful, i looked at the Dataquest
solution and rebuild it. My problem above was, that my if clause didnt count how many characters at any title were out of the ascii range. So even if only 1 "®" was there, the whole datapoint was marked as faulty. 
Now above i will iterate over every character of the title string. When there are more than 3 characters out of the interval 0 - 127 in ASCII code, the title will be deleted. Key function is the ord() function, which returns the ASCII number of a character. **google_clean -> clean_roman | apple_clean -> clean_roman**

In [5]:
def roman_letters_only(dataset,title_column):
    titles=[value[title_column] for value in dataset] # extrahiere nur die "App" Column um nicht das ganze Dataset zu "for loopen index bleibt gleich"
    global clean_roman
    clean_roman=[]
    for i, string in enumerate(titles):        
        x=0
        for char in string:                
            if ord(char)<127 and ord(char)>0:
                x+=1
            if x>len(string)-4:
                clean_roman.append(dataset[i])
                break

roman_letters_only(google_unique,0)
google_unique_roman=clean_roman
roman_letters_only(apple_unique,1)
apple_unique_roman=clean_roman

print("Length GOOGLE: ", len(google_unique_roman))
print("Length APPLE: ", len(apple_unique_roman))

print(apple_unique_roman[:3])


Length GOOGLE:  9614
Length APPLE:  6184
[['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic'], ['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'], ['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']]


# Subset "Free" Apps
Free apps are marked the following: 
Apple price == 0.0 Index Column 4
Google Type == "Free" Index Column 6

In [6]:
def only_free(dataset,free_column):
    global free_apps
    free_apps=[dataset[0]]
    for data in dataset:
        if data[free_column]=="0.0" or data[free_column]== "Free":
            free_apps.append(data)
            #print(data[free_column])
            
    #print(free_apps)
            
only_free(google_unique_roman,6)
google_unique_roman_free=free_apps
only_free(apple_unique_roman,4)
apple_unique_roman_free=free_apps
print("GOOGLE:",len(google_unique_roman_free))
print("APPLE:",len(apple_unique_roman_free))
print(google_unique_roman_free[:2],"\n")
print(apple_unique_roman_free[:2])

GOOGLE: 8861
APPLE: 3223
[['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'], ['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']] 

[['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic'], ['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']]


# Install Value Dictionary
To eazily apply thresholds on my Datasets, i wrote this code, to create a dictionary, which for example returns float 1 000 000 for the key "1M"

In [7]:
liste=[]
installs_dict={}
for data in google_unique_roman_free[1:]:
    if data[5] not in liste:
        liste.append(data[5])

liste.sort(reverse=True)
for value in liste:
    value=value.replace("+","")
    value=value.replace(",","")
    if "000000000" in value:
        value2=value.replace("000000000","")
        
        installs_dict[value2+"B"]=float(value)
    elif "000000" in value:
        value2=value.replace("000000","")
        
        installs_dict[value2+"M"]=float(value)
        
    elif "000" in value:
        value2=value.replace("000","")
        
        installs_dict[value2+"k"]=float(value)
    else:
        
        installs_dict[value]=float(value)

print(installs_dict)

print(installs_dict["1M"])


{'500M': 500000000.0, '500k': 500000.0, '500': 500.0, '50M': 50000000.0, '50k': 50000.0, '50': 50.0, '5M': 5000000.0, '5k': 5000.0, '5': 5.0, '100M': 100000000.0, '100k': 100000.0, '100': 100.0, '10M': 10000000.0, '10k': 10000.0, '10': 10.0, '1B': 1000000000.0, '1M': 1000000.0, '1k': 1000.0, '1': 1.0, '0': 0.0}
1000000.0


# Visualize our data
Now we've cleaned our data. Now i create a frequency tables of the "Genres" / "prime_genre" of each dataset.  
In easy words: **Which app Genre has the most apps with high download values.** 
Then we'll see which genre is most succesfull in each App-Store. 

For our next steps, we will know, what genre our new app should be in, to have a higher success probability. 
If our new app is profitable in the google Playstore after six months, we'll code the same app for the AppStore. 
Google "Genres" Column Index: 9
Apple "prime_genre" Column Index: 11

In [8]:
##my own function##
def frequency_table(dataset,index):
    freq_dict={}
    l=len(dataset)-1 # Header -1
    for data in dataset[1:]:
        genre=data[index]
        if genre in freq_dict:
            freq_dict[genre]+=1
        else:
            freq_dict[genre]=1
    for key, value in freq_dict.items():
        freq_dict[key]=round((value/l)*100,2) # dict values in percent
        
    return freq_dict






##this function is given from Dataquest##
def display_table(dataset, index):
    table = frequency_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

print("Google Categorys :")
display_table(google_unique_roman_free,1)
print("\n Google Genres :")
display_table(google_unique_roman_free,9)
print("\n Apple Prime_Genres :")
display_table(apple_unique_roman_free,11)


Google Categorys :
FAMILY : 18.93
GAME : 9.7
TOOLS : 8.45
BUSINESS : 4.59
LIFESTYLE : 3.91
PRODUCTIVITY : 3.89
FINANCE : 3.7
MEDICAL : 3.52
SPORTS : 3.4
PERSONALIZATION : 3.32
COMMUNICATION : 3.24
HEALTH_AND_FITNESS : 3.08
PHOTOGRAPHY : 2.95
NEWS_AND_MAGAZINES : 2.8
SOCIAL : 2.65
TRAVEL_AND_LOCAL : 2.34
SHOPPING : 2.25
BOOKS_AND_REFERENCE : 2.14
DATING : 1.86
VIDEO_PLAYERS : 1.79
MAPS_AND_NAVIGATION : 1.4
FOOD_AND_DRINK : 1.24
EDUCATION : 1.17
ENTERTAINMENT : 0.96
LIBRARIES_AND_DEMO : 0.94
AUTO_AND_VEHICLES : 0.93
HOUSE_AND_HOME : 0.82
WEATHER : 0.8
EVENTS : 0.71
PARENTING : 0.65
ART_AND_DESIGN : 0.64
COMICS : 0.62
BEAUTY : 0.6

 Google Genres :
Tools : 8.44
Entertainment : 6.07
Education : 5.35
Business : 4.59
Productivity : 3.89
Lifestyle : 3.89
Finance : 3.7
Medical : 3.52
Sports : 3.47
Personalization : 3.32
Communication : 3.24
Action : 3.1
Health & Fitness : 3.08
Photography : 2.95
News & Magazines : 2.8
Social : 2.65
Travel & Local : 2.33
Shopping : 2.25
Books & Reference : 2.14

# Analyzing our data
**Google Apps:**

As we can see, in the category column, "Family" is by far the most frequent (18.93 %). Then "Game and "Tools" categories with around the same (9.7,8.45 %) frequence come next. 

In the "Genres" Column, the results are way more near together. The first 13 apps are between 8.44% and 3.08%.
The top three are Tools (8.44%), Entertainment (6.07%) and Education (5.35%)

**Apple Apps**

Here we only have the "prime_genre" column to look at. Here we can make a interesting observation. 58.16% of all free Apps are belong to the genre "Games". After that way behind comes on place 2 and 3: Entertainment(7.88%) and Photo & Video(4.97%). When we compare that to the 9.7% of the "Game" Category of Google, we see that the share of Game-Apps is much larger in the Apple App-Store. 

**Observations:** 

When we compare that to the 9.7% of the "Game" Category of Google, we see that the share of Game-Apps is much larger in the Apple App-Store. Also we can see that on both the datasets, the Genre "Entertainment" is in both Datasets the second largest share. 

Now we see how many apps exist in each Category/Genre. But that still doesnt tell us, if the Category/App is succesfull / has a lot of very succesfull Apps with lots of installs.. 
We will fix this in the next codeblocks. 

# Google Dataset Visualisation
Here we can see how the share of categories/genres change, the more succesfull and widely adopted the app gets. 
At first glance we see that Smartphones are used the most for Communication and as a replacement for Tools (~25% of all installs are Apps in those Categories)
Apps installed over 100 mil times, are 25% Communication Apps and 13.64% "Tools" Apps. When the installs are above 500 million, the top 2 Categories/Genres are "Communication" (30%) and "Social" (15%). We see, the most widely known and used apps, are Communication Apps and social Apps. 

In [10]:
#####Installs per Genre#########
def installs_genre(dataset,index,index2):
    print("Google:")
    genre_dict={}
    sum_installs=0
    genre_sorted=[]
    for data in dataset[1:]:
        genre=data[index2]
        installs=data[index].replace("+","")
        installs=installs.replace(",","")
        installs=float(installs)
        sum_installs+=installs
        if genre in genre_dict:
            genre_dict[genre]+=installs
        else: 
            genre_dict[genre]=installs
    for value in genre_dict:
        genre_tuple=(genre_dict[value],value)
        genre_sorted.append(genre_tuple)
    genre_sorted.sort(reverse=True)
    for x in genre_sorted:
        prop=round((x[0]/sum_installs)*100,2)
        print(prop,"% of alls Apps are of the Genre",x[1],"- Total installs:",x[0])
installs_genre(google_unique_roman_free,5,9)

def installs_genre(dataset,index,index2):
    print("\nApple:")
    genre_dict={}
    sum_installs=0
    genre_sorted=[]
    for data in dataset[1:]:
        genre=data[index2]
        installs=data[index].replace("+","")
        installs=installs.replace(",","")
        installs=float(installs)
        sum_installs+=installs
        if genre in genre_dict:
            genre_dict[genre]+=installs
        else: 
            genre_dict[genre]=installs
    for value in genre_dict:
        genre_tuple=(genre_dict[value],value)
        genre_sorted.append(genre_tuple)
    genre_sorted.sort(reverse=True)
    for x in genre_sorted:
        prop=round((x[0]/sum_installs)*100,2)
        print(prop,"% of alls Apps are of the Genre",x[1],"- Total installs:",x[0])
installs_genre(apple_unique_roman_free,5,11)


threshhold0=installs_dict["1M"]  ##Intervall von treshhold0 bis treshhold1 wird ausgewertet## 
threshhold1=900000000000000
def frequency_table_genre(dataset,index,index2):
    print("\n")
    freq_dict={}
    for data in dataset[1:]:
        genre=data[index]
        installs=data[index2].replace("+","")
        installs=installs.replace(",","")
        installs=float(installs)
        if genre in freq_dict and installs>threshhold0 and installs<threshhold1:
            freq_dict[genre]+=1
        elif installs>threshhold0 and installs<threshhold1:
            freq_dict[genre]=1
            
    k=sum(freq_dict.values())
    for key, value in freq_dict.items():
        freq_dict[key]=round((value/k)*100,2) # dict values in percent
    
                
    return freq_dict

def display_table(dataset, index, index2):
    table = frequency_table_genre(dataset, index, index2)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

print("\n ~Google~ Share of Genres in the Intervall:",threshhold0,"to",threshhold1,"installs")
display_table(google_unique_roman_free,9,5)
print("\n~Google~ Share of Categories in the Intervall:",threshhold0,"to",threshhold1,"installs")
display_table(google_unique_roman_free,1,5)

#######Installs : % of Apps in that popularity range#######
print("\n")
def frequency_table_genre(dataset,index2):
    freq_dict={}
    for key,value in installs_dict.items():
        freq_dict[key]=0
        for data in dataset[1:]:
            installs=data[index2].replace("+","")
            installs=installs.replace(",","")
            installs=float(installs)
            if installs == value and key in freq_dict:
                freq_dict[key]+=1
            
    k=sum(freq_dict.values())
    for key, value in freq_dict.items():
        if value !=0:
            freq_dict[key]=round((value/k)*100,2) # dict values in percent
    
                
    return freq_dict

def display_table(dataset,index2):
    table = frequency_table_genre(dataset,index2)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1]+"+", ':', entry[0],"%")

print("~Google~ Percentage of Apps installed certain number of times")
print("times installed : % of all Apps\n")
display_table(google_unique_roman_free,5)

#####To spectate the Apps even further#####

n=installs_dict["500M"]
k=installs_dict["1B"]
def apps_counter(dataset,index):
    print("\n")
    counter=0
    for data in dataset[1:]:
        installs=data[index].replace("+","")
        installs=installs.replace(",","")
        installs=float(installs)
        if installs>=n and installs<k:
            counter+=1
            
    print("There are",counter,"Apps, which are installed more than",n,"and less than",k,"times\n")
    
apps_counter(google_unique_roman_free,5)

def apps_name_printer(dataset,index,index2):
    for data in dataset[1:]:
        name=data[index2]
        installs=data[index].replace("+","")
        installs=installs.replace(",","")
        installs=float(installs)
        if installs>=n and installs<k:
            print(name)
            
    
apps_name_printer(google_unique_roman_free,5,0)

######Proportion of the most installed Apps######
def apps_proportion(dataset,index,index2): #index install column , index2 app_name column
    print("\n")
    name_l=[]
    sum_installs=0
    for data in dataset[1:]:
        installs=data[index].replace("+","")
        installs=installs.replace(",","")
        installs=float(installs)
        name=data[index2]
        t=(installs,name)
        name_l.append(t)
        sum_installs+=installs
    name_l.sort(reverse=True)
    print("Sum of installs:",sum_installs)
    stopper=0
    for x in name_l:
        prop=round((x[0]/sum_installs)*100,2)
        if stopper>=50:
            break
        else:
            print(x[1],prop,"% of all installs")
            stopper+=1
    
apps_proportion(google_unique_roman_free,5,0)
        

Google:
14.7 % of alls Apps are of the Genre Communication - Total installs: 11036906201.0
10.64 % of alls Apps are of the Genre Tools - Total installs: 7991043474.0
7.71 % of alls Apps are of the Genre Productivity - Total installs: 5791629314.0
7.31 % of alls Apps are of the Genre Social - Total installs: 5486861902.0
6.19 % of alls Apps are of the Genre Photography - Total installs: 4647268815.0
5.22 % of alls Apps are of the Genre Video Players & Editors - Total installs: 3916731720.0
5.0 % of alls Apps are of the Genre Arcade - Total installs: 3753691940.0
4.62 % of alls Apps are of the Genre Action - Total installs: 3465986940.0
4.05 % of alls Apps are of the Genre Casual - Total installs: 3042798570.0
4.01 % of alls Apps are of the Genre Entertainment - Total installs: 3014302513.0
3.86 % of alls Apps are of the Genre Travel & Local - Total installs: 2894604086.0
3.15 % of alls Apps are of the Genre News & Magazines - Total installs: 2368196260.0
2.22 % of alls Apps are of the G

1M+ : 15.73 %
100k+ : 11.56 %
10M+ : 10.52 %
10k+ : 10.2 %
1k+ : 8.4 %
100+ : 6.92 %
5M+ : 6.84 %
500k+ : 5.58 %
50k+ : 4.77 %
5k+ : 4.51 %
10+ : 3.54 %
500+ : 3.25 %
50M+ : 2.29 %
100M+ : 2.12 %
50+ : 1.92 %
5+ : 0.79 %
1+ : 0.51 %
500M+ : 0.27 %
1B+ : 0.23 %
0+ : 0.05 %


There are 24 Apps, which are installed more than 500000000.0 and less than 1000000000.0 times

Google Duo - High Quality Video Calls
imo free video calls and chat
LINE: Free Calls & Messages
UC Browser - Fast Download Private & Secure
Candy Crush Saga
Pou
My Talking Tom
Temple Run 2
Facebook Lite
Snapchat
Google Translate
SHAREit - Transfer & Share
Gboard - the Google Keyboard
Microsoft Word
Dropbox
Google Calendar
Cloud Print
MX Player
Twitter
Flipboard: News For Our Time
Clean Master- Space Cleaner & Antivirus
Viber Messenger
Samsung Health
Security Master - Antivirus, VPN, AppLock, Booster


Sum of installs: 75076551335.0
YouTube 1.33 % of all installs
WhatsApp Messenger 1.33 % of all installs
Subway Surfers 1.33

# Conclusion
For the first App i recommend a App in the Genre of "Communication" or "Tools". These are the most used Apps from Users in the Google Playstore. Also Apps from the Genre "Social Networking" are the most installed on Apple Devices. This support my recommendation for an App of the genre "Communication". 